sdk/cosmos/azure_data_cosmos/src/models/item.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::Model;
+use serde::de::DeserializeOwned;
+
+/// Returned by Cosmos DB APIs that return a single item.
+///
+/// In some circumstances, an API that _can_ return an item will **not** return an item.
+/// For example, you can use [`ItemOptions`](crate::options::ItemOptions) to configure APIs
+/// that write new or updated items to avoid returning the updated item.
+// Marking this non-exhaustive would make things very difficult for users.
+// We encourage them to pattern match against it, but it's not clear how to deal with the "other" case required by being non_exhaustive.
+// This isn't an enum representing a value returned by the server, so we wouldn't add things to this enum later.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub enum Item<T> {
+    // Why do we have our own enum and not use Option<T>?
+    // We need something we can return from SDK APIs, which means it has to implement azure_core::Model.
+    // It's not reasonable to implement azure_core::Model on Option<T> for any T that's deserializable, so we have our own enum.
+    // This also allows us to use names that are more relevant to the scenario
+    /// Indicates that the server returned the item content.
+    Present(T),
+
+    /// Indicates that the server omitted the item content in the response,
+    /// as directed by the [`ItemOptions`](crate::options::ItemOptions) used when making the request
+    #[default]
+    Omitted,
+}
+
+impl<T> Item<T> {
+    /// Returns a boolean indicating if the [`Item<T>`] was omitted.
+    pub fn is_omitted(&self) -> bool {
+        match self {
+            Item::Present(_) => false,
+            Item::Omitted => true,
+        }
+    }
+
+    /// Returns a boolean indicating if the [`Item<T>`] is present.
+    pub fn is_present(&self) -> bool {
+        match self {
+            Item::Present(_) => true,
+            Item::Omitted => false,
+        }
+    }
```

Comment:
Are these more useful in some scenarios that just a match as with `Option<T>` e.g.,

```rust
let Present(thing) = item else {
    // return or whatever
}
```

If you think these are generally useful without creating an anti-pattern as I've been seeing with newer rust devs in related projects (checking `Option<T>::is_some()`, for example, keep them. We just generally don't want to add public APIs for "might be useful" scenarios vs. "most likely or definitely useful" so we don't end up owning them forever. Too many APIs can also lead to confusion.